File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> count(n, 0);
        for (auto& edge : edges) {
            count[edge[1]] = 1;
        }
        int m = 0;
        for (int i = 0; i < n; i++) {
            if (!count[i]) {
                count[m++] = i;
            }
        }
        count.resize(m);
        return count;
    }
};

File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        l = [0 for i in range(n)]
        for edge in edges:
            l[edge[1]] = 1
        result = []
        for i in range(n):
            if l[i] == 0:
                result.append(i)
        return result